lua_api.txt: more detailed documentation of 'set_attach' method.

Andrey2470T opened the pull request:
After some discussions on Discord and in #11582 which relate to my misunderstanding of the objects attachment mechanism, I found it necessary to finally document about it not to repeat my mistakes in future. Correct me though, if I wrote something wrong.

## To do

This PR is Ready for Review.

## How to test
Just read the changes in the only file.

sfan5 commented:
This paragraph sounds good to me but its grammar should be fixed and the "multiply coordinate by 10" should not be called a  *bug*.

SmallJoker commented:
This text would fit well into a modding guide. lua_api.txt is an API reference, not a tutorial.

Andrey2470T commented:
@sfan5 which a grammar should be fixed? I didn't see any mistakes or typos in the text. This is a bug because it is necessary to multiply position by 10 and also it was confirmed by rubenwardy on Discord.

Andrey2470T commented:
So can it be merged now or how? It is just impossible to review long-term as there is no even a code, just an extension for the API documentation.

sfan5 commented on doc/lua_api.txt:
```diff
@@ -6432,6 +6432,25 @@ Doing so is much less error-prone and you will never need to wonder if the
 object you are working with still exists.
 
 
+### Attachment
+
+It is possible to attach objects to other objects (`set_attach` method).
+
+When an object is attached, its position gets relative to the parent`s one and
+rotation is a sum of its own, the parent`s and relative rotations. Standard
+methods like 'set_pos' and 'set_rotation' won`t work, you should call repeatedly
+`set_attach` with already new position or rotation tables.
+
+**Bug**: keep in mind to multiply relative position by 10 when attaching to make it actually proper!
+
+Example:
+`child:set_attach(parent, "", vector.multiply(vector.subtract(child:get_pos(), parent:get_pos()), 10))`.
+
+It is also possible to set bones for parent objects like in Blender (`set_bone_position` method)
+and use them for child ones to adjust their position/rotation. After assignment a bone to an attached object,
+it will follow movement and rotation of that bone.
```
```suggestion
### Attachments

It is possible to attach objects to other objects (`set_attach` method).

When an object is attached, it is positioned relative to the parent's position and rotation.
`get_pos` and `get_rotation` will always return the parent's values and can't be set directly.
To change position or rotation call `set_attach` again with the new values.

**Note**: Just like model dimensions, the relative position in `set_attach` must be multiplied by 10.

It is also possible to attach to a bone of the parent object (as defined in the model).
After assignment a bone to an attached object, the child will follow movement and rotation of that bone.
```

Andrey2470T replied:
I think it is necessary to mention also the child rotation is a sum of its own, the parent's snd relative rotations, at least I checked it experimentally.

sfan5 commented:
That's better but grammar is still wrong in a few places, apostrophes are `'` not <code>\`</code>, the example does not make much sense (if objects are already attached their position will be exactly equal) and the reference to model dimensions is gone.

Andrey2470T commented:
Added references to the model dimensions in two places, replaced the apostrophes to the single quotes. I think the example is important, because I show how the relative position should be set correctly for the child. And can you elaborate in which sentences grammar is wrong?

appgurueu commented:
Should be reduced to a note on `set_attach` / `get_attach` stating (roughly) the following:
> Position, rotation and scale become relative to the parent:
> * Attachment position (which must be given in model space, which is block space scaled up by a factor of 10 times parent visual scale) is added to the parent (bone) position
> * Analogeously, attachment rotation is added to parent (bone) rotation
> * Child visual scale is multiplied by parent visual scale

Andrey2470T commented:
@appgurueu if it is better to make as notes for `set_attach` / `get_attach`, I think then it is just necessary to supplement already the existent text of those methods. The whole potential text would look like so:
`
* `set_attach(parent[, bone, position, rotation, forced_visible])`
      * `bone`: string.
           * This is a name of the parent's bone that is set with `set_bone_position`.
           * After setting, the child will follow the movement and rotation of that bone.
           * Default is `""`, the root bone.
      * `position`: `{x=num, y=num, z=num}`.
           * Relative to the parent's model space.
           * Default is `{x=0, y=0, z=0}`.
           **Note**: Just like model dimensions, it should be multiplied by 10 to set it actually proper.
      * `rotation`: `{x=num, y=num, z=num}`.
           * Rotation around each axis in degrees.
           * Default is `{x=num, y=num, z=num}`.
           * This rotation is added to a sum of rotations of the parent and the child.
      * `forced_visible`: Boolean to control whether the attached entity should appear in first person. Default `false`. * This command may fail silently (do nothing) when it would result in circular attachments.
`

So ok?

Andrey2470T commented:
@appgurueu So, what do you think about my suggested changes to the attachment doc above?

appgurueu commented:
> @appgurueu So, what do you think about my suggested changes to the attachment doc above?

Yes

Andrey2470T commented:
Done.

Andrey2470T commented:
@appgurueu @sfan5 are your objections/suggestions about this PR? Can it be merged now?

rubenwardy commented on doc/lua_api.txt:
```diff
@@ -6484,13 +6483,22 @@ object you are working with still exists.
-       should appear in first person. Default `false`.
-    * This command may fail silently (do nothing) when it would result
+	* `bone`: string.
+		* This is a name of the parent's bone that is set with `set_bone_position`.
```
isn't it from the model?

rubenwardy commented on doc/lua_api.txt:
```diff
@@ -6484,13 +6483,22 @@ object you are working with still exists.
+		* Relative to the parent's model space.
+		* Default is {x=0, y=0, z=0}.
+		* Note: Just like model dimensions, it should be multiplied by 10
+		  to set it actually proper.
```
```suggestion
		 compared to a world position.
```

rubenwardy commented on doc/lua_api.txt:
```diff
@@ -6484,13 +6483,22 @@ object you are working with still exists.
+		* Note: Just like model dimensions, it should be multiplied by 10
+		  to set it actually proper.
+	* `rotation`: {x=num, y=num, z=num}.
+		* Rotation around each axis in degrees.
```
```suggestion
		* Rotation around each axis in degrees, relative to parent's rotation.
```
